### ISSUE TITLE
Bug: false positive for `match must be exhaustive` when enum uses feature flags 

### ISSUE THREAD

**Issue by ozwaldorf** [intellij-rust/intellij-rust#9910]
## Environment

* **IntelliJ Rust plugin version:** `0.4.185.5086-223`
* **Rust toolchain:** on `nightly`, `stable`, `1.65.0`
* **IDE name and version:** Clion 
* **Operating system:** Manjaro

## Problem description
When using an enum that is populated using feature flags, and there is only a single entry, matching for that entry incorrectly reports that there are uncaught cases. Adding a `_ => unreachable!()` clause as suggested by clion to the match then causes clippy and check to error that it's unreachable code because the enum only has a single type. 

## Steps to reproduce

(`rust_libp2p::identity::Keypair`)
```rs
#[derive(Debug, Clone)]
#[allow(clippy::large_enum_variant)]
pub enum Keypair {
    /// An Ed25519 keypair.
    Ed25519(ed25519::Keypair),
    /// An RSA keypair.
    #[cfg(all(feature = "rsa", not(target_arch = "wasm32")))]
    Rsa(rsa::Keypair),
    /// A Secp256k1 keypair.
    #[cfg(feature = "secp256k1")]
    Secp256k1(secp256k1::Keypair),
    /// An ECDSA keypair.
    #[cfg(feature = "ecdsa")]
    Ecdsa(ecdsa::Keypair),
}
```

And the match on said enum with no features enabled:
```rs
match self {
  Keypair::Ed25519(keypair) => {}
}
```
